4.assess_fibrosis_reversal/scripts/0.apply_models.py

```diff
@@ -0,0 +1,259 @@
+parquet_file = BytesIO(response.content)
+
+# Load the Parquet file into a pandas dataframe
+df = pd.read_parquet(parquet_file)
+
+# Get the column names that do not start with "Metadata"
+model_features = [col for col in df.columns if not col.startswith("Metadata")]
+
+# Output the number of columns to check if it matches correctly
+print(len(model_features))
+
+
+# ## Set paths and variables
+
+# In[5]:
+
+
+# Directory for probability data to be saved
+prob_dir = pathlib.Path("./prob_data")
+prob_dir.mkdir(exist_ok=True)
+
+# Directory with normalized plate datasets  
+data_dir = pathlib.Path("../3.preprocessing_features/data/single_cell_profiles")
+
+# Use rglob to search for files with the suffix *_sc_normalized.parquet
+parquet_files = list(data_dir.rglob("*_sc_normalized.parquet"))
+
+# Load all matching parquet files into a list of DataFrames
+dfs = [pd.read_parquet(file) for file in parquet_files]
+
+# Print the file paths and check the content if needed
+for file in parquet_files:
+    print(f"Loaded file: {file}")
+
+
+# ## Filter the data for the model features and add to dictionary to process
+
+# In[6]:
+
+
+# Dictionary to store plate data
+plate_data_dict = {}
+
+# Loop through and process each parquet file
+for parquet_file in parquet_files:
+    # Extract the plate name from the file name
+    plate_name = parquet_file.name.split('_')[0]
+    print(f"Processing plate: {plate_name}")
+
+    # Load the Parquet file
+    df = pd.read_parquet(parquet_file)
+
+    # Drop rows with NaN values in feature columns that the model uses
+    df = df.dropna(subset=model_features)
+
+    # Capitalize the cell type values to match the model
+    df["Metadata_cell_type"] = df["Metadata_cell_type"].str.capitalize()
+
+    # Extract metadata columns
+    metadata_columns = [col for col in df.columns if col.startswith("Metadata_")]
+
+    # Extract feature columns that don't start with "Metadata_"
+    feature_columns = [col for col in df.columns if not col.startswith("Metadata_")]
+
+    # Filter columns in the data frame to only include those in the model
+    filtered_feature_columns = [col for col in feature_columns if col in model_features]
+
+    # Filter the DataFrame to keep only the desired columns
+    model_df = df[metadata_columns + filtered_feature_columns]
+
+    # Store the processed DataFrame in the dictionary under the key "model_df"
+    plate_data_dict[plate_name] = {"model_df": model_df}
+
+    # Print info about the processed DataFrame
+    print(f"Number of unique treatments in {plate_name}: {df['Metadata_treatment'].nunique()}")
+    print(f"Shape of the model DataFrame: {model_df.shape}")
+
+
+# ## Extract final model predicted probabilities for each treatment
+
+# In[7]:
+
+
+# Create an empty DataFrame to store the results
+combined_prob_df = pd.DataFrame()
+
+# Loop through each model in the models directory
+for model_path in models_dir.iterdir():
+    model_type = model_path.stem.split("_")[5]  # Get the model type
```

Comment:
In this case, it is hardcoded based on the format we established in the previous repo. So technically, this should not change unless we go back to the repo and change stuff for the manuscript which is possible.

For now, I will keep this since this is directly related to what we have already generated. It is as great point though.